src/js/contentAwareMiddleware.js

```diff
@@ -0,0 +1,85 @@
+/*
+
+    A router which passes a request to different handlers based on what the request accepts.  See the documentation for
+    details:
+
+    https://github.com/GPII/gpii-express/blob/master/docs/contentAwareRouter.md
+
+ */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+require("./requestAwareMiddleware");
+
+fluid.registerNamespace("gpii.express.middleware.contentAware");
+
+/**
+ * 
+ * Handle all traffic routed to us.  Locate an appropriate handler if possible, return a clear error if not.
+ * 
+ * @param that {Object} The router component itself.
+ * @param request {Object} The Express {Request} object.
+ * @param response {Object} The Express {Response} object.
+ * 
+ */
+gpii.express.middleware.contentAware.delegateToHandler = function (that, request, response, next) {
+    var handlerGrades = gpii.express.middleware.contentAware.getHandlerGradesByContentType(that, request);
+    if (handlerGrades) {
+        that.events.onRequest.fire(request, response, handlerGrades);
+    }
+    else {
+        next({ isError: true, message: that.options.messages.noHandlerFound });
+    }
+};
+
+/**
+ * 
+ * Return the first matching handler that matches the `request` object's `accept` header.  Handlers are sorted by 
+ * priority before they are evaluated.
+ * 
+ * @param that {Object} The router component itself.
+ * @param request {Object} The Express {Request} object.
+ * @returns {Array} An array of handler grades that will be used to create our handler.
+ * 
+ */
+gpii.express.middleware.contentAware.getHandlerGradesByContentType = function (that, request) {
+    var orderedHandlers = gpii.express.orderByPriority(that.options.handlers);
+    var matchingHandler = fluid.find(orderedHandlers, function (value) {
+        return Boolean(value.contentType && request.accepts(value.contentType)) ? value : undefined;
+    });
+
+    return matchingHandler ? matchingHandler.handlerGrades : undefined;
+};
+
+fluid.defaults("gpii.express.middleware.contentAware", {
+    gradeNames: ["gpii.express.middleware"],
+    timeout: 5000,
+    messages: {
+        noHandlerFound: "Could not find an appropriate handler for the content types you accept."
+    },
+    distributeOptions: {
+        source: "{that}.options.timeout",
+        target: "{that gpii.express.handler}.options.timeout"
+    },
+    events: {
+        onRequest: null
+    },
+    dynamicComponents: {
```

Comment:
Yes, I would have assumed it would work until I had clear evidence to the contrary.  Wrapping options in an enclosing object didn't help.  Anyway, I have only to uncomment a few lines to step through it together in our next pairing session.